tests/lib/rules/component-tags-order.js

```diff
@@ -102,11 +104,90 @@ tester.run('component-tags-order', rule, {
       output: null,
       options: [{ order: ['docs', 'script', 'template', 'style'] }]
     },
+    {
+      code: '<script setup></script><script></script><template></template><style></style>',
+      output: null,
+      options: [{ order: ['script[setup]', 'script', 'template', 'style'] }]
+    },
+    {
+      code: '<template></template><script setup></script><script></script><style></style>',
+      output: null,
+      options: [{ order: [['script[setup]', 'script', 'template'], 'style'] }]
+    },
+    {
+      code: '<script></script><script setup></script><template></template><style></style>',
+      output: null,
+      options: [{ order: ['script', 'script[setup]', 'template', 'style'] }]
```

Comment:
I think we probably need to test that `'script'` also applies to `<script setup>`.

```suggestion
      options: [{ order: ['script', 'template', 'style'] }]
```